### PR TITLE
Feature/95 free text unit w md support

### LINKED ISSUE #95: Free text unit with markdown support
# User Story

### As a: 

teacher

### I want to

... add free text units (in a text area) with markdown support

### so that

I can conveniently add and edit free text content of my sections.

## Acceptance criteria:

- [x] I can choose "free text" when adding a new unit in the section configuration 
- [x] A text area is presented that gives me markdown support
- [x] The markdown input is displayed in proper HTML format on the students frontend

## Additional info:

Live preview would be nice but is be no means required.

### PR DESCRIPTION
**Pull request by aeimer**
# Request Title:
Adds FreeText Units

## Description:
- Adds FreeTextUnits.
  - API / DB
  - FE
- Some small Bugfixes

resolves #95 

## Improvements
- Uptimerobot Badge
- Style in docker.sh
- Adds Freetext Unit

## Known Issues:
- The styling is not optimal, should improve that
- A preview Tab, in GH style, would be nice (we should enable previewing for the whole lecture)

### CODE REVIEW

**Comment by aeimer** [comment]
The only thing we should fix in the future, is to style the html generated by the markdown-parser

File: api/src/controllers/FreeTextUnitController.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import passportJwtMiddleware from '../security/passportJwtMiddleware';
+import {UnitController} from './UnitController';
+import {FreeTextUnit} from '../models/units/FreeTextUnit';
+import {IFreeTextUnit} from '../../../shared/models/units/IFreeTextUnit';
+
+@JsonController('/units/free-texts')
+@UseBefore(passportJwtMiddleware)
+export class FreeTextUnitController extends UnitController {
+
+  @Post('/')
```
Comment (grosspersky) [inline api/src/controllers/FreeTextUnitController.ts]: Add ``@Authorized('teacher')`` here.

Comment (aeimer) [inline api/src/controllers/FreeTextUnitController.ts]: I've added this, do I need to retest this?

Comment (grosspersky) [inline api/src/controllers/FreeTextUnitController.ts]: This should work without additional testing, after you add the missing import for the ``@Authorized``-Directive.

**Pull request merged**